bids/layout/layout.py

```diff
@@ -957,8 +957,11 @@ def get(self, return_type='object', target=None, scope='all',
         # Strip leading periods if extensions were passed
         if 'extension' in filters:
             # XXX 0.14: Disable drop_dot option
-            drop_dot = (self.config['bids'].entities['extension'].pattern ==
-                        '[._]*[a-zA-Z0-9]*?\\.([^/\\\\]+)$')
+            drop_dot = (
+                len(self.config) == 1
+                and self.config[next(iter(self.config))].entities['extension'].pattern
+                == '[._]*[a-zA-Z0-9]*?\\.([^/\\\\]+)$'
+            )
```

Comment:
I don't understand the `len == 1` and `next` stuff... What about the following?:

```suggestion
            drop_dot = (
                'bids' in self.config and
                self.config['bids'].entities['extension'].pattern == '[._]*[a-zA-Z0-9]*?\\.([^/\\\\]+)$'
            )
```

Reply:
the reason behind the logic was to preserve the original behavior on any config, not just the default (`bids`). I was testing on the templateflow client, since 0.11 broke it. That is using a config with the same criteria for extensions as the default config (https://github.com/templateflow/python-client/blob/e4280b5df18e305279fd1d95d46f7660450a689d/templateflow/conf/config.json#L70-L72), but it would fall into the other half of the if/else block and alter the query.

 the `next(iter(...))` stuff is just to select the first (and only) item in the dictionary, open to suggestions if you know of a better way.